src/vs/vscode.proposed.d.ts

```diff
@@ -1168,16 +1177,18 @@ declare module 'vscode' {
 	 */
 	export class CustomExecution2 {
 		/**
-		 * @param process The [TerminalVirtualProcess](#TerminalVirtualProcess) to be used by the task to display output.
+		 * @param process The [Pseudotrminal](#Pseudoterminal) to be used by the task to display output.
 		 * @param callback The callback that will be called when the task is started by a user.
 		 */
-		constructor(callback: (thisArg?: any) => Thenable<TerminalVirtualProcess>);
+		constructor(callback: (thisArg?: any) => Thenable<Pseudoterminal>);
 
 		/**
-		 * The callback used to execute the task. Cancellation should be handled using the shutdown method of [TerminalVirtualProcess](#TerminalVirtualProcess).
-		 * When the task is complete, onDidExit should be fired on the TerminalVirtualProcess with the exit code with '0' for success and a non-zero value for failure.
+		 * The callback used to execute the task. Cancellation should be handled using
+		 * [Pseudoterminal.shutdown](#Pseudoterminal.shutdown). When the task is complete,
+		 * [Pseudoterminal.onDidExit](#Pseudoterminal.onDidExit) should be fired with the exit code
+		 * with '0' for success and a non-zero value for failure.
```

Comment:
Are negative numbers allowed? If not,m can we comment that here as well.

Reply:
This part hasn't checked, I'm leaning towards just making this an `Event<void>` though and letting the extension show a notification if that's what it wants to do when it fails.